src/state.js

```diff
@@ -0,0 +1,98 @@
+const { ReplaySubject, BehaviorSubject } = require('rxjs');
+
+const { transaction, log } = require('./reports');
+const { feeRate, balance } = require('./config');
+
+const initialState = {
+  balance: {
+    // base coin, in USDT-BTC, this is USDT
+    base: balance[0],
+    // target coin, in USDT-BTC, this is BTC
+    target: balance[1]
+  },
+  fees: 0,
+  baseLine: 0,
+  mean: 0,
+  hold: false
+};
+
+const actions = {
+  hold(state) {
+    return Object.assign({}, state, { hold: true });
+  },
+  release(state) {
+    return Object.assign({}, state, { hold: false });
+  },
+  updateBase(state, action) {
+    return Object.assign({}, state, { baseLine: action.payload });
+  },
+  updateMean(state, action) {
+    return Object.assign({}, state, { mean: action.payload });
+  },
+  // action payload is the rate to trade at
+  bid(state, action) {
+    const tradeRate = action.payload;
+    const operation = state.balance.base / tradeRate;
+    const fee = operation * feeRate;
+    const result = operation - fee;
+    const feeInBase = fee * tradeRate;
+
+    transaction({
+      isBid: true,
+      tradeRate, result, feeInBase,
```

Comment:
transaction method does not receive a property called feeInBase, it receives a property called fee